### PR TITLE
Invalid calcuting span idx on default and core commands

### PR DESCRIPTION
**Pull request by Mehrbod2002**
Fixes #9072 

The problem is indexing an invalid span size on some core command but not default command and function command

### CODE REVIEW

File: crates/nu-parser/src/parser.rs [hunk 1]
```diff
@@ -592,7 +592,14 @@ fn calculate_end_span(
...
                 && spans.len() > (signature.required_positional.len() - positional_idx)
             {
-                spans.len() - (signature.required_positional.len() - positional_idx - 1)
+                spans.len()
+                    - (signature.required_positional.len()
+                        - positional_idx
+                        - (if signature.category == Category::Core {
+                            1
+                        } else {
+                            2
```
Comment (sholderbach) [inline crates/nu-parser/src/parser.rs:601]: I don't think we can make this general assumptions that all commands that get labelled with `Category::Core` would share this behavior and should rather fix where their parsing is misbehaving.

Comment (Mehrbod2002) [reply crates/nu-parser/src/parser.rs:601]: The mistake is the whole signature. Can manipulate the core command signature on positional argument .

Comment (Mehrbod2002) [reply crates/nu-parser/src/parser.rs:601]: The second condition should be disabled for default command.

Comment (sholderbach) [reply crates/nu-parser/src/parser.rs:601]: From this change it is really hard to follow how it relates to custom command calls failing during syntax highlighting with more than a certain number of positional arguments but it not being a problem for lower numbers. As described by #9072 

Based on the diff it just appears to add another special case for which it is unclear why it needs to be there that will make future bugfixes an even greater nightmare.

To close #9072 we should also include a test that demonstrates the parser doesn't panic in this circumstance.

Comment (Mehrbod2002) [reply crates/nu-parser/src/parser.rs:601]: Please attention to the issue ,
When you insert a args on cli it panics , not when the whole command executes . How can I add tests with it then? 
Review the signature some command like `if` , in if there are multi span which are greater than size of positional arg , but when I create function `def` it attach a signature which has span_idx that only should increase when it wants to calculate the index.

**Comment by sholderbach** [comment]
I am closing this attempt as the proposed patch may break as soon as certain commands get recategorized.

Furthermore it seems that this is merely stopping something from going catastrophic in a certain circumstance, there seems to be a correctness problem in another part of the codebase, which required you to apply this hack.
